core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -254,11 +254,26 @@ private void handleRequest(ChannelHandlerContext ctx, DecodedHttpRequest req) th
         final String hostname = hostname(ctx, headers);
         final VirtualHost host = config.findVirtualHost(hostname);
 
+        final PathMappingContext mappingCtx = DefaultPathMappingContext.of(host, hostname,
+                                                                           path, query, headers,
+                                                                           host.producibleMediaTypes());
         // Find the service that matches the path.
-        final PathMapped<ServiceConfig> mapped = host.findServiceConfig(path, query);
+        final PathMapped<ServiceConfig> mapped;
+        try {
+            mapped = host.findServiceConfig(mappingCtx);
+        } catch (Throwable cause) {
+            if (cause instanceof ServiceUnavailableException) {
+                respond(ctx, req, HttpStatus.SERVICE_UNAVAILABLE);
```

Comment:
You're right. I missed the fact that they are children of `HttpResponseException`. :-( 
I fixed here and the other one similar to this.